pyomo/repn/plugins/lp_writer.py

```diff
@@ -0,0 +1,594 @@
+class LPWriter(object):
+    CONFIG = ConfigBlock('lpwriter')
+    CONFIG.declare(
+        'show_section_timing',
+        ConfigValue(
+            default=False,
+            domain=bool,
+            description='Print timing after writing each section of the LP file',
+        ),
+    )
+    CONFIG.declare(
+        'skip_trivial_constraints',
+        ConfigValue(
+            default=False,
+            domain=bool,
+            description='Skip writing constraints whose body is constant',
+        ),
+    )
+    CONFIG.declare(
+        'file_determinism',
+        ConfigValue(
+            default=FileDeterminism.ORDERED,
+            domain=InEnum(FileDeterminism),
+            description='How much effort to ensure file is deterministic',
+            doc="""
+            How much effort do we want to put into ensuring the
+            LP file is written deterministically for a Pyomo model:
+                NONE (0) : None
+                ORDERED (10): rely on underlying component ordering (default)
+                SORT_INDICES (20) : sort keys of indexed components
+                SORT_SYMBOLS (30) : sort keys AND sort names (not declaration order)
+            """,
+        ),
+    )
+    CONFIG.declare(
+        'symbolic_solver_labels',
+        ConfigValue(
+            default=False,
+            domain=bool,
+            description='Write variables/constraints using model names',
+            doc="""
+            Export variables and constraints to the LP file using human-readable
+            text names derived from the corresponding Pyomo component names.
+            """,
+        ),
+    )
+    CONFIG.declare(
+        'row_order',
+        ConfigValue(
+            default=None,
+            description='Preferred constraint ordering',
+            doc="""
+            List of constraints in the order that they should appear in the
+            LP file.  Unspecified constraints will appear at the end.""",
+        ),
+    )
+    CONFIG.declare(
+        'column_order',
+        ConfigValue(
+            default=None,
+            description='Preferred variable ordering',
+            doc="""
+
+
+            List of variables in the order that they should appear in
+            the LP file.  Note that this is only a suggestion, as the LP
+            file format is row-major and the columns are inferred from
+            the order in which variables appear in the objective
+            followed by each constraint.""",
+        ),
+    )
+    CONFIG.declare(
+        'labeler',
+        ConfigValue(
+            default=None,
+            description='Callable to use to generate symbol names in LP file',
+            doc="""
+            Export variables and constraints to the LP file using human-readable
+            text names derived from the corresponding Pyomo component names.
+            """,
+        ),
+    )
+    CONFIG.declare(
+        'output_fixed_variable_bounds',
```

Comment:
It would write out unused variables to the LP file (with their bounds).  Since the v2 writer never goes and collects variables, it can't (easily or reliably) support this "feature".

Reply:
Got it.